5.2 release notes (#31)
hazelcast/hazelcast-platform-operator-docs: docs/modules/ROOT/pages/release-notes.adoc

= 5.2 Release Notes

> In this release, we introduced Backup & Restore to/from PVC and hostPath

Reviewer: ```suggestion
In this release, we introduced the Backup and Restore feature with support for PVC and hostPath. This feature uses xref:5.1@hazelcast:storage:persistence.adoc[Hazelcast Persistence] under the hood.
```

> features.

Reviewer: ```suggestion
```

== New features

> * Backup data into

Reviewer: ```suggestion
* Back up data into
```

https://kubernetes.io/docs/concepts/storage/persistent-volumes/[PVC]
(via

> https://docs.hazelcast.com/hazelcast/latest/storage/persistence[Hazelcast

Reviewer: ```suggestion
xref:5.1@hazelcast:storage:persistence.adoc[Hazelcast
```

We should use internal links for docs pages. Please apply this change to other docs links.

Persistence])
+

> The persistence feature needs to be enabled at Hazelcast CR:

Reviewer: We have docs for all these features. It's better to link to the doc than provide instructions in release notes. This applies to all instructions in this section.

Author: They are not instructions actually, give some idea/starting points about possible YAML usage to users. Providing some details to customer/users via release notes is a good idea, I think.

Reviewer: In that case, we should consider the following:
- Add a link to the feature's doc so that users can learn more about how it works/how to use it.
- Avoid duplication by including the YAML snippets like we do here: https://github.com/hazelcast/hazelcast-platform-operator-docs/edit/main/docs/modules/ROOT/pages/backup-restore.adoc

Author: Good points, I will consider them during the preparation of 5.3 release notes :man_shrugging:

  scheduling:
    topologySpreadConstraints:
      - maxSkew: 1
        topologyKey: kubernetes.io/hostname
        whenUnsatisfiable: DoNotSchedule
        labelSelector:
          matchLabels:
            app.kubernetes.io/instance: hazelcast
....
* Auto
https://docs.hazelcast.com/hazelcast/latest/storage/triggering-force-start[Force
Start] support. The Platform Operator to detect failed Hazelcast members
and automatically trigger a force-start
+
....
apiVersion: hazelcast.com/v1alpha1
kind: Hazelcast
metadata:
  name: hazelcast
spec:
  ...
  licenseKeySecret: hazelcast-license-key
  persistence:
    baseDir: "/data/persistence/"
    clusterDataRecoveryPolicy: "FullRecoveryOnly"
    autoForceStart: true
....
* ImagePullSecrets configuration for Hazelcast and Management Center CRs
* Management Center can be started with existing PV claim:
+
....
apiVersion: hazelcast.com/v1alpha1
kind: ManagementCenter
metadata:
  name: managementcenter
spec:
  ...
  persistence:
    enabled: true
    existingVolumeClaimName: existing-pvc
....
* Pod scheduling support for Management Center CR.
* `topologySpreadConstraints` scheduling support for Hazelcast and
Management Center CRs.

== Changes

> * Upgrade Hazelcast and MC versions to 5.1.x

Reviewer: Do we need to update this attributes to the new versions?

- https://github.com/hazelcast/hazelcast-platform-operator-docs/blob/340cef55aaad9059a0d4528284937f6735f66cc2/docs/antora.yml#L15
- https://github.com/hazelcast/hazelcast-platform-operator-docs/blob/340cef55aaad9059a0d4528284937f6735f66cc2/docs/antora.yml#L16

Author: They will be updated by release workflow at operator repo:
https://github.com/hazelcast/hazelcast-platform-operator/blob/main/.github/workflows/docs-release.yaml

(https://github.com/hazelcast/hazelcast-platform-operator/pull/148)
* Improve Operator Logging and add Ginkgo v2 CLI for better test
verbosity
(https://github.com/hazelcast/hazelcast-platform-operator/pull/151)

> * Detailed member statuses

Author: Both of them are good points. But this detailed member status enhancement is not the real one, we added it for the internal mechanism. With the 5.3 release, we will release the full member state(like MC did). So I think for now we can keep it like that.

  Members:
    Connected:  false
    Ip:         10.164.2.23
    Message:    back-off 10s restarting failed container=hazelcast pod=hazelcast-0_default(07c1a692-1be9-408d-b245-bb93cf01af66)
    Pod Name:   hazelcast-0
    Reason:     CrashLoopBackOff
  Message:      multiple (1) errors: pod hazelcast-0 in namespace default failed for back-off 10s restarting failed container=hazelcast pod=hazelcast-0_default(07c1a692-1be9-408d-b245-bb93cf01af66):
....
+
Successful one:
+
....
Status:
  Hazelcast Cluster Status:
    Ready Members:  3/3
  Members:
    Connected:  true
    Ip:         10.164.2.27
    UID:        85eeb6f3-00a3-4bed-aa4c-30959d6bda67
    Version:    5.0.2
    Connected:  true
    Ip:         10.164.1.12
    UID:        e546754e-209d-4ac5-9fd0-119a6d8bf371
    Version:    5.0.2
    Connected:  true
    Ip:         10.164.0.14
    UID:        5a7b1a1f-4a1c-47ff-8151-24db0e5db51e
    Version:    5.0.2
  Phase:        Running
....
* Show external IP of Hazelcast members and Management Center URL at CR
statuses:
+
....
$ kubectl get hazelcast
NAME               STATUS    MEMBERS   EXTERNAL-ADDRESSES
hazelcast  Running   3/3         <EXTERNAL_IP>:5701
....

== Fixed Issues

> * Add support for asynchronous Hazelcast connection with reconnection

Reviewer: We should describe the issue in fixes so that users understand what the fix means. Without that context, fixes are often meaningless.

For this example:

'When a cluster failed to restart for any reason, the Go client failed to reconnect to it even after the cluster successfully restarted. We've now added support for the Go client to asynchronously reconnect to a cluster.'

Author: you are right, good feedback 👍 I will update the PR
